src/kixi/event_fixer.clj

```diff
@@ -57,32 +91,30 @@
                                 (hour->s3-prefix hour)
                                 nil)))
   ([s3-base-dir prefix marker]
-   (let [list-objects-res (s3/list-objects (merge {:bucket-name s3-base-dir
-                                                   :prefix prefix
-                                                   :max-keys max-objects}
-                                                  (when marker
-                                                    {:marker marker})))]
+   (let [list-objects-res (s3/list-objects credentials (merge {:bucket-name s3-base-dir
+                                                                  :prefix prefix
+                                                                  :max-keys max-objects}
+                                                                 (when marker
+                                                                   {:marker marker})))]
      (concat (:object-summaries list-objects-res)
-             (when (:next-marker list-objects-res)
-               (hour->s3-object-summaries s3-base-dir prefix (:next-marker list-objects-res)))))))
+                (when (:next-marker list-objects-res)
+                  (hour->s3-object-summaries s3-base-dir prefix (:next-marker list-objects-res)))))))
 
 (defn object-summary->local-file
   [s3-base-dir local-base-dir]
   (fn
     [s3-object-summary]
-    (let [s3-object (s3/get-object :bucket-name s3-base-dir
-                                   :key (:key s3-object-summary))
-          ^File local-file (io/file local-base-dir (last (string/split (:key s3-object-summary) #"/")))]
+    (let [^File local-file (io/file local-base-dir (last (string/split (:key s3-object-summary) #"/")))]
       (when-not (.exists local-file)
         (do (.createNewFile local-file)
-            (io/copy (:object-content s3-object)
-                     local-file)))
+            (let [s3-object (s3/get-object credentials :bucket-name s3-base-dir
+                                           :key (:key s3-object-summary))
+                  in (:input-stream s3-object)]
+              (io/copy in
+                       local-file)
+              (. in close))))
```

Comment:
use `with-open` instead to make sure things get closed if there is an exception (or put this in a `finally`)

Reply:
Done